src/app/components/map.tsx

```diff
@@ -0,0 +1,48 @@
+"use client";
+
+import "mapbox-gl/dist/mapbox-gl.css";
+import mapboxgl, { GeoJSONSource } from "mapbox-gl";
+import { useEffect, useRef } from "react";
+
+export default function MapComponent() {
+  const mapContainer = useRef<HTMLDivElement>(null);
+
+  if (!process.env.NEXT_PUBLIC_MAPBOX_API_TOKEN) {
```

Comment:
The way I've been taught by mapbox's documentation to use this component is by wrapping it in a useEffect and making use of useRef. This means this page needs to be a client component. Using a client component means your environment variable needs to be public.

I have a feeling the environment variable situation might have something to do with the failed deployment by Vercel?

Reply:
Yep that's exactly it, unfortunately 'cause I'm on a free account I'll have to go and add the env variable myself. That's ok tho, we'll hop in a call or sommat and set it up